app/_components/_sections/SavedWorkoutsList.tsx

```diff
@@ -0,0 +1,196 @@
+"use client";
+
+import React, { useState, useEffect } from "react";
+import { FaArrowLeftLong } from "react-icons/fa6";
+import { CiBookmarkMinus } from "react-icons/ci";
+import { CgNotes } from "react-icons/cg";
+import { HiMiniSquares2X2 } from "react-icons/hi2";
+import { FaSpinner } from "react-icons/fa";
+import DeleteWorkoutModal from "../modals/deleteWorkoutModal";
+import { useTranslation } from "react-i18next";
+import { useDeleteWorkout } from "@/app/_services/userService";
+
+interface Workout {
+  _id: string;
+  image_url: string;
+  name: string;
+  level: string;
+  duration?: number;
+  workout_schedule: any[];
+}
+
+interface SavedWorkoutsModalProps {
+  workouts: Workout[];
+  onClose: () => void;
+  refetchSavedWorkouts: () => void;
+}
+
+const calculateTotalDuration = (workout: Workout) => {
```

Comment:
Issue: The function runs on every render; it could be memoized.
Solution: Use useMemo to cache the result unless the workout data changes.

For example: 
```
const totalDurationMinutes = useMemo(() => calculateTotalDuration(workout), [workout]);
```